client/src/components/RequestDataWrapper/RequestDataWrapper.tsx

```diff
@@ -1,20 +1,17 @@
-import React, { FunctionComponent } from 'react';
+import React, { PropsWithChildren } from 'react';
 
-// Types
-import { LocationInfo, WeatherInfo } from '../../types';
-
-interface RequestDataWrapperProps {
-  data: LocationInfo | WeatherInfo | null;
-  loading: boolean;
-  error: boolean | string | GeolocationPositionError | null;
+interface RequestDataWrapperProps<T> {
+  data?: T | T[];
+  loading?: boolean;
+  error?: boolean | string | GeolocationPositionError | null;
 }
 
-const RequestDataWrapper: FunctionComponent<RequestDataWrapperProps> = ({
+const RequestDataWrapper = <T,>({
```

Comment:
wasn't you able to use `T extends unknown` or something like that?

Reply:
I don't know. I've read eslint rules again, and it says that it's unnecessary, because it is extended to unknown by default. (https://typescript-eslint.io/rules/no-unnecessary-type-constraint/). If I have to add here in this component, then would I have to add same syntax to other places where I've used generics as well?

Reply:
you need to add something after <T to help compiler understand that it's a typescript syntax, not jsx. i still recommend to use `extends unknown` for this purpose because it doesn't cause guesses about haven't anything missed after comma sign